### PR TITLE
Merge remaining commits of 4.7.0 and 4.7.1 into master

### PR DESCRIPTION
**Pull request by saturninoabril**
Merge remaining commits of 4.7.0 and 4.7.1 into master.

(Note: There were some commits included here which were not squashed in release-4.7).

### CODE REVIEW

File: app/team_test.go [hunk 1]
```diff
@@ -460,3 +460,92 @@ func TestAddUserToTeamByHashMismatchedInviteId(t *testing.T) {
...
+
+	if _, err := th.App.CreateTeam(team); err != nil {
+		t.Log(err)
+		t.Fatal("Should create a new team")
+	}
+
+	maxUsersPerTeam := th.App.Config().TeamSettings.MaxUsersPerTeam
+	defer func() {
+		th.App.UpdateConfig(func(cfg *model.Config) { cfg.TeamSettings.MaxUsersPerTeam = maxUsersPerTeam })
+		th.App.SetDefaultRolesBasedOnConfig()
```
Comment (jespino) [inline app/team_test.go]: This line was accidentally included (I don't know how the tests passes in the branch where it was merged).

Comment (AndersonWebStudio) [reply app/team_test.go]: @jespino anything I can do to help with this one?

Comment (jespino) [reply app/team_test.go]: Not really, we only need to remove that line, I can enter in the branch and make the commit.

Comment (jespino) [reply app/team_test.go]: Done!, I have removed the line, I'm not sure if this will fix the merge, but at least, will fix one of the problems.

**Pull request merged**